Xamarin.Forms.Core/BaseExportRendererAttribute.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+
+namespace Xamarin.Forms.Core
+{
+	[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+	public abstract class BaseExportRendererAttribute : HandlerAttribute
+	{
+		#region MinimumSdkVersion
+		int _minimumSdkVersion = int.MinValue;
+		public int MinimumSdkVersion
+		{
+			get { return _minimumSdkVersion; }
+			set
+			{
+				if (_minimumSdkVersion == value)
+					return;
+
+				_minimumSdkVersion = value;
+
+				if (_minimumSdkVersion > MaximumSdkVersion)
+					throw new ArgumentException("Minimum SDK version must be less than or equal to the maximum SDK version.");
+			}
+		}
+		#endregion
+
+		#region MaximumSdkVersion
+		int _maximumSdkVersion = int.MaxValue;
+		public int MaximumSdkVersion
+		{
+			get { return _maximumSdkVersion; }
+			set
+			{
+				if (_maximumSdkVersion == value)
+					return;
+
+				_maximumSdkVersion = value;
+
+				if (_maximumSdkVersion < MinimumSdkVersion)
+					throw new ArgumentException("Maximum SDK version must be greater than or equal to the minimum SDK version.");
+			}
+		}
+		#endregion
+
+		protected BaseExportRendererAttribute(Type handler, Type target) : base(handler, target)
+		{
+		}
+
+		public override bool ShouldRegister()
+		{
+			// Most custom renderers won't use versioning
+			if (MinimumSdkVersion == int.MinValue && MaximumSdkVersion == int.MaxValue)
+				return base.ShouldRegister();
+
+			// Conditional registration is being used
+			var majorVersion = GetMajorVersion();
+
+			if (MinimumSdkVersion <= majorVersion && majorVersion <= MaximumSdkVersion)
+				return base.ShouldRegister();
+
+			return false;
+		}
+
+		public abstract int GetMajorVersion();
```

Comment:
How about get only property? It can be also overridden in sub class.

[Choosing Between Properties and Methods](https://msdn.microsoft.com/en-us/library/ms229054(v=vs.100).aspx)

Reply:
I'll get back to this later. I think the property would need to be abstract in the base since we'll need polymorphic behavior in the base `ShouldRegister` method.